core/timeMeasure.go

```diff
@@ -0,0 +1,112 @@
+package core
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type timeMeasure struct {
+	mut         sync.Mutex
+	identifiers []string
+	started     map[string]time.Time
+	elapsed     map[string]time.Duration
+}
+
+// NewTimeMeasure returns a new timeMeasure instance used to measure duration between finished and started events
+func NewTimeMeasure() *timeMeasure {
+	return &timeMeasure{
+		identifiers: make([]string, 0),
+		started:     make(map[string]time.Time),
+		elapsed:     make(map[string]time.Duration),
+	}
+}
+
+// Start marks a start event for a provided identifier
+func (tm *timeMeasure) Start(identifier string) {
+	tm.mut.Lock()
+	_, hasStarted := tm.started[identifier]
+	_, hasElapsed := tm.elapsed[identifier]
+	if !hasStarted && !hasElapsed {
+		tm.identifiers = append(tm.identifiers, identifier)
+	}
+
+	tm.started[identifier] = time.Now()
+	tm.mut.Unlock()
+}
+
+func (tm *timeMeasure) addIdentifier(identifier string) {
+	_, hasStarted := tm.started[identifier]
+	if hasStarted {
+		return
+	}
+
+	_, hasElapsed := tm.elapsed[identifier]
+	if hasElapsed {
+		return
+	}
+
+	tm.identifiers = append(tm.identifiers, identifier)
+}
+
+// Finish marks a finish event for a provided identifier
+func (tm *timeMeasure) Finish(identifier string) {
+	tm.mut.Lock()
+	defer tm.mut.Unlock()
+
+	timeStarted, ok := tm.started[identifier]
+	if !ok {
+		return
+	}
+
+	tm.elapsed[identifier] += time.Now().Sub(timeStarted)
+	delete(tm.started, identifier)
+}
+
+// GetMeasurements returns a logger compatible slice of interface{} containing pairs of (identifier, duration)
+func (tm *timeMeasure) GetMeasurements() []interface{} {
+	data, newIdentifiers := tm.GetContainingDuration()
+
+	tm.mut.Lock()
+
+	output := make([]interface{}, 0)
+	for _, identifier := range newIdentifiers {
+		duration := data[identifier]
+		output = append(output, identifier)
+		output = append(output, fmt.Sprintf("%.2fs", duration.Seconds()))
```

Comment:
I think we can use at least 4 decimals for each duration. 
Also, you can rename it to MeasurementsLoggerFormat

Reply:
done